src/components/spectrogram/spectrogram.ts

```diff
@@ -439,9 +450,14 @@ export class SpectrogramComponent extends SignalWatcher(AbstractComponent(LitEle
     return invalidationKeys.some((key) => change.has(key));
   }
 
-  private frameBufferTime(): number {
-    const frameIndex = new Float32Array(this.currentTimeBuffer)[0];
-    return frameIndex;
+  /**
+   * Returns the amount of time processed by the high accuracy time processor
+   * It is important to note that the time does not represent the actual time
+   * of the audio element, but is actually a very accurate amount of time that
+   * has passed since the worklet node started processing audio
+   */
+  private highAccuracyElapsedTime(): number {
+    return new Float32Array(this.currentTimeBuffer)[0];
```

Comment:
don't recast the buffer each time you access it. store a reference to the typed array view